lib/base/perfdatavalue.cpp

```diff
@@ -45,6 +45,8 @@ PerfdataValue::Ptr PerfdataValue::Parse(const String& perfdata)
 
 	String valueStr = perfdata.SubStr(eqp + 1, spq - eqp - 1);
 
+	valueStr = valueStr.ToLower();
+
 	size_t pos = valueStr.FindFirstNotOf("+-0123456789.e");
```

Comment:
Have you tried adding upper-case `E` here? If it doesn't work then, I'd consider this a bug in `Convert::ToDouble`.

Reply:
Adding the upper-case `E` works.
```diff
diff --git a/lib/base/perfdatavalue.cpp b/lib/base/perfdatavalue.cpp
index 6371b54ef..5531dc418 100644
--- a/lib/base/perfdatavalue.cpp
+++ b/lib/base/perfdatavalue.cpp
@@ -45,9 +45,7 @@ PerfdataValue::Ptr PerfdataValue::Parse(const String& perfdata)
 
        String valueStr = perfdata.SubStr(eqp + 1, spq - eqp - 1);
 
-       valueStr = valueStr.ToLower();
-
-       size_t pos = valueStr.FindFirstNotOf("+-0123456789.e");
+       size_t pos = valueStr.FindFirstNotOf("+-0123456789.eE");
 
        double value = Convert::ToDouble(valueStr.SubStr(0, pos));
 
@@ -162,7 +160,7 @@ String PerfdataValue::Format() const
 
 Value PerfdataValue::ParseWarnCritMinMaxToken(const std::vector<String>& tokens, std::vector<String>::size_type index, const String& description)
 {
-       if (tokens.size() > index && tokens[index] != "U" && tokens[index] != "" && tokens[index].FindFirstNotOf("+-0123456789.e") == String::NPos)
+       if (tokens.size() > index && tokens[index] != "U" && tokens[index] != "" && tokens[index].FindFirstNotOf("+-0123456789.eE") == String::NPos)
                return Convert::ToDouble(tokens[index]);
        else {
                if (tokens.size() > index && tokens[index] != "")
```

Which proposal should I implement?

Reply:
I'd prefer the one that updates the string. Bonus points for making it a constant as it is there in two places.

With the original PR, I guess `tokens[index] != "U"` in the if would always evaluate to true. So maybe another reason to do it the other way? (Note to self/thinking out loud: how would this break and do we want a test-case that covers this?)